contracts/src/v0.8/keystone/KeystoneForwarder.sol

```diff
@@ -133,17 +129,19 @@ contract KeystoneForwarder is IForwarder, ConfirmedOwner, TypeAndVersionInterfac
     bytes calldata reportContext,
     bytes[] calldata signatures
   ) external nonReentrant {
-    if (rawReport.length < REPORT_METADATA_LENGTH) {
+    if (rawReport.length < METADATA_LENGTH) {
       revert InvalidReport();
     }
 
-    (bytes32 workflowId, uint32 donId, bytes32 workflowExecutionId, address workflowOwner) = _getMetadata(rawReport);
+    (bytes32 workflowExecutionId, uint32 donId /* uint32 donConfigVersion */, , bytes2 reportId) = _getMetadata(
+      rawReport
+    );
 
     // f can never be 0, so this means the config doesn't actually exist
     if (s_configs[donId].f == 0) revert InvalidDonId(donId);
 
-    bytes32 reportId = _reportId(receiverAddress, workflowExecutionId);
-    if (s_reports[reportId].transmitter != address(0)) revert ReportAlreadyProcessed(reportId);
+    bytes32 combinedId = _combinedId(receiverAddress, workflowExecutionId, reportId);
```

Comment:
Inconsistent naming
```suggestion
    bytes32 messageId = _combinedId(receiverAddress, workflowExecutionId, reportId);
```

I would prefer transmissionId. We can fix in a follow-up.